app/controllers/versions_controller.rb

```diff
@@ -0,0 +1,37 @@
+class VersionsController < ApplicationController
+  include Helpers::Geminabox
+
+  helper_method :link_to_gem, :reverse_dependencies, :load_gems
+
+  def show
+    # Taken from https://github.com/geminabox/geminabox/blob/0.12.4/lib/geminabox/server.rb#L108-L109
+    gems = Hash[server_instance.send(:load_gems).by_name]
+    @gem = gems[params['gemname']]
+    @version = @gem && @gem.find { |g| g.number = params['version'] }
```

Comment:
`==` ?
also maybe better to do `@gem = server_instance.send(:load_gems).detect { |g| g.name == name && g.version == version}`